### ISSUE TITLE
Synchronize images/videos to proper Android folders

### ISSUE THREAD

**Issue by Quix0r** [nextcloud/android#1131]
### Actual behaviour
Currently, for example all images are being downloaded somewhere outside the "gallery" app's folder and they are not showing up there. If you want to make them visible, you have manually copy them from that hidden directory (Android/data/com.nextcloud.app or so?) to the gallery which duplicates storage usage.

### Expected behaviour
They should be "accessible by the default "Gallery" app so people can instantly view them after syncing them.
 
### Steps to reproduce
1. Simply sync a file from nextcloud to your android.
2. It won't show up in Gallery app or any proper default app.
3. The app Cirrus by the way is working that way.

### Environment data
Android version:

Device model:
Samsung J5

Stock or customized system:
Default Android

Nextcloud app version:
_latest stable from Play store_

Nextcloud server version:
11.0.2 (stable)

**Comment by Fensterbank** [comment]
I had exactly the same issue on my phone, while my new installation on my tablet works like you want.

When I synchronize a folder with my nextcloud on my tablet and this folder contains images or videos, it will be shown in the gallery! On my smartphone this wasn't the case.

I realized a new installation created a folder like `sdcard\nextcloud\user@hostname`, which will be indexed by your android gallery. My old installation on the smartphone - even after the app was updated - holds the data inside of `Android/data/com.nextcloud.client` like you said. And this folder won't be indexed by your gallery.

**tl;dr:**
You have to uninstall and reinstall the nextcloud client on your android, a nextcloud root folder will be created on your sdcard or device storage and your gallery will have access to it.
I just tested it and can confirm it on my phone.

**Comment by Quix0r** [comment]
Okay, will test it.

**Comment by tobiasKaminsky** [comment]
I am not sure if this is 1.4.3 but it will be in next version. There it will be again in "/sdcard/media.../" to allow indexing.

**Comment by Fensterbank** [comment]
But will already installed apps move their whole content to the new folder structure?
In the current version, indexing works with the current data location, but not when the app has been installed in the past using the old folder structure inside `com.nextcloud.client`.

Uninstalling and reinstalling the client is a working workarround, but it's still not very cool.

**Comment by tobiasKaminsky** [comment]
Settings -> storage location and then changing it to the desired path should be sufficient.

**Comment by tobiasKaminsky** [comment]
Closing this one as it is already merged in master.
If this problem persists in next version, please reply.

**Comment by Quix0r** [comment]
Sorry, had been busy in the last weeks. :-(

**Comment by DaveBurkhardt** [comment]
I'm being pessimistic here; but I have yet to see the sync work.  At this point in time I'd not let the app delete the files.  The sync is broken for a lot of us.  very unreliable (doesn't work).

Hopefully you are one of the luck ones where the sync's coding hooks work.

I have literally 100's (maybe 1000's) of photo's videos for months now which are only on my families local SD cards because I upgraded to latest version of this app.

Considering going to ownclouds app as this apps sync code is not very robust at this point in time.
